### ISSUE TITLE
) expected ts-styled-plugin(9999)

### ISSUE THREAD

**Issue by Josema** [microsoft/typescript-styled-plugin#100]
```
    background-image: url(${p =>
        p.disabled ? 'assets/img/button-grey.png' : p.image});
```

![image](https://user-images.githubusercontent.com/1819888/57127594-31850f00-6d91-11e9-9941-66cdcc94938e.png)

If I disable the whole plugin using `compilerOptions` the code colors is a mesh.

**Comment by mjbvz** [comment]
Please share the text of the code along with screenshots

**Comment by danielbayerlein** [comment]
This code produces the problem.

```js
const getPadding = ({ right, large }) => css`
  padding-${right ? 'left' : 'right'}: ${large ? 24 : 12}px;
  padding-${right ? 'right' : 'left'}: ${large ? 64 : 48}px;
`
```

**Comment by WilliamIPark** [comment]
Any solution to this? 

Can reproduce with:
```javascript
const AddTeamMember = styled("button")`
  background-image: url(${props =>
    props.negative ? AddUserIconLight : AddUserIconDark});
`
```

**Comment by tannerbaum** [comment]
I also have been getting this error with the following code:
```js
.click-to-play-overlay {
	position: absolute;
        // ... 
	background: url(${(props: { thumbnail: string }) =>
		props.thumbnail ? props.thumbnail : `${process.env.BASE_URL}/br24_logo_map.png`});
	}
```

I tried to write a test for this but couldn't get it to fail. Maybe I am misunderstanding how the tests and/or code works?
 
```js
    it('should not error for prettified multi-line prop usage', async () => {
        const errorResponse = await getSemanticDiagnosticsForFile(
            `let css: any = {};
            const googleUrl = 'https://www.google.com';
            const props = {
                thumbnail: ''
            }; 
            const ListNoteTitle = css.button\`
            background: url($\{(props: { thumbnail: string }) =>
				props.thumbnail ? props.thumbnail : \`\${googleUrl}/google.png\`});
            \`;`);

        assert.isTrue(errorResponse.success);
        assert.strictEqual(errorResponse.body.length, 0);
    });
```

**Comment by mjbvz** [comment]
Where did you put the test? It should be in https://github.com/Microsoft/typescript-styled-plugin/blob/78e7ad681145211a9e955498bf5389a1ea697a3d/e2e/tests/errors.js

I get 3 errors if I add that test

**Comment by tannerbaum** [comment]
@mjbvz really? When I added it and ran the e2e script I got 34 tests failing but that wasn't one of them 😆 Maybe something is wrong with my setup. 

But if the test is failing for you then I hope that helps

**Comment by mjbvz** [comment]
The tests seem to be running correctly on travis so you likely have a setup issue. Make sure you run `npm install` inside the `e2e` folder too before running the tests themselves

I do not have time to fix this issue; it needs to come from the community

**Comment by publicJorn** [comment]
If you have the same issue, you can work around it like this:

```js
const elem = styled.div`
  border-radius: 10px;
  background-image: url(${(p) =>
      p.available ? availableIcon : unavailableIcon});
`
```

becomes:

```js

const elem = styled.div`
  border-radius: 10px;
  ${(p) =>
      p.available
        ? `background-image: ${availableIcon};`
        : `background-image: ${unavailableIcon};`}
`
```

**Comment by karlhorky** [comment]
Wonder if [my error here](https://github.com/microsoft/typescript-styled-plugin/issues/88#issuecomment-544107132) is related / the same issue:

```js
import { createGlobalStyle, ThemeProvider } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'Averta Standard';
    font-weight: 800;
    font-style: normal;
    font-display: swap;
    src: local('☺'),
      url('${
        process.env.PUBLIC_URL
      }/averta-standard-extra-bold.woff2') format('woff2');
  }
```

<img src=https://user-images.githubusercontent.com/1935696/67138979-0c0a9580-f24b-11e9-830a-199e22a69f1d.png />

**Comment by MilanKrupa** [comment]
**I kind of found a solution** for this issue in vscode. Just install vscode-styled-components in Diego Lincoln's version instead of Julien Poissonier's. This version has typescript-styled-plugin removed and it works perfect for me.

[Link to fixed version](https://marketplace.visualstudio.com/items?itemName=diegolincoln.vscode-styled-components#review-details)

**Comment by karlhorky** [comment]
@MilanKrupa Oh, interesting. Does this still work with TypeScript? What are the downsides?

@dilincoln are you thinking of submitting these changes to Julien's repo as pull requests?

**Comment by dilincoln** [comment]
Hi @karlhorky,

> @MilanKrupa Oh, interesting. Does this still work with TypeScript? What are the downsides?
Works with .js .ts .tsx .vue files.

> @dilincoln are you thinking of submitting these changes to Julien's repo as pull requests?
I will make a pull request after add more features and snippets.

**Comment by karlhorky** [comment]
I guess the downsides of the approach with snippets are:

- the autocompletion does not come from Intellisense
- you cannot get extra information about CSS properties by hovering

Maybe it's better to stick with `typescript-styled-plugin` and try to fix it instead.

**Comment by dilincoln** [comment]
@karlhorky thanks for the observation. I will improve the extension

**Comment by dilincoln** [comment]
Hey guys, i have published a new version with: 

- added `typescript-styled-plugin` support
- auto add colon and semicolon improved

See how it's working now:

![demo](https://user-images.githubusercontent.com/37885454/92606159-0665d600-f289-11ea-9ea5-29b3e088f338.gif)

**Comment by mjbvz** [comment]
@dilincoln Can you contribute the fixes back to this plugin / the styled extension that most people are already using?

**Comment by dilincoln** [comment]
@mjbvz of course, first i need to add auto colon/comma for react-native and use the extension for some time to avoid to make a pull request with unknown bugs

**Comment by ricardobossan** [comment]
https://github.com/microsoft/typescript-styled-plugin/issues/100 this solved the issue for me.

**Comment by ricardobossan** [comment]
https://gitmemory.com/issue/microsoft/typescript-styled-plugin/100/572014717 this solved it for me.

**Comment by kmvan** [comment]
> **I kind of found a solution** for this issue in vscode. Just install vscode-styled-components in Diego Lincoln's version instead of Julien Poissonier's. This version has typescript-styled-plugin removed and it works perfect for me.
> 
> [Link to fixed version](https://marketplace.visualstudio.com/items?itemName=diegolincoln.vscode-styled-components#review-details)

works.

**Comment by MilanKrupa** [comment]
> works.

I'm glad it still does 'cause I've struggled quite a while on this one.

**Comment by ra-kesh** [comment]
> > works.
> 
> I'm glad it still does 'cause I've struggled quite a while on this one.

Not sure why it's not working for me .  have disabled the other extension , should I uninstall it completely ???

**Comment by jjnilton** [comment]
Lincoln's version didn't fix the issue for me on VSCode 1.60.2, so revert to Poissonnier's (1.6.6).

Related code:

```
const selectAllOnClick = keyframes`
  to {
    -webkit-user-select: text;
    user-select: text;
  }
`;

const animation = (props) => css`
  ${selectAllOnClick} 100ms step-end forwards
`;
```

**Comment by Speckope** [comment]
The following code also produces a similar error.

![image](https://user-images.githubusercontent.com/57017082/140417150-51fa6033-8502-42ec-bb9a-d0d6a187a995.png)
![image](https://user-images.githubusercontent.com/57017082/140417174-73a0b609-b1ca-4416-867f-d83888fa9625.png)

**Comment by Justin9606** [comment]
I solved it by creating the jconfig.json inside the root of my project and adding:

if your project is in Typescript then create tsconfig.json*

{
"compilerOptions": {
"plugins": [
{
"name": "typescript-styled-plugin",
"lint": {
"validProperties": [
"margin-vertical",
"margin-horizontal",
"padding-vertical",
"padding-horizontal",
"shadow-color",
"shadow-opacity",
"shadow-radius",
"elevation"
]
}
}
]
}
}

**Comment by knitkode** [comment]
This code produces an error too:

```jsx
const TableBodyCell = styled.td`
  width: 14%;
`;

const TableBodyRow = styled.tr`
  &:nth-child(even) ${TableBodyCell}:nth-child(odd),
  &:nth-child(odd) ${TableBodyCell}:nth-child(even) {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.07);
  }
  
  &:first-child ${TableBodyCell} {
    border-top: 0;
  }
`;
```

**Comment by SUCHMOKUO** [comment]
also error:
<img width="404" alt="图片" src="https://user-images.githubusercontent.com/30554294/182041609-5750fcec-9880-470e-9e9c-098fba510def.png">

**Comment by hungdoansy** [comment]
This code produces an error too:

```jsx
const MenuFlyoutBrowserStyle = css`
  min-width: unset;
  right: -8px;

  ${ExternalNavMenuItem}:nth-child(1),
  ${NavMenuItem}:nth-child(1) {
    padding-top: 0.75rem;
  }
`

const MenuFlyoutMobileStyle = css`
  ${ExternalNavMenuItem}:nth-child(1),
  ${NavMenuItem}:nth-child(1) {
    padding-top: 0.75rem;
  }
`
```

**Comment by zquancai** [comment]
<img width="639" alt="Screen Shot 2023-01-13 at 17 46 36" src="https://user-images.githubusercontent.com/8609783/212289466-53a3880d-2c27-4ac0-baf7-b3319a237350.png">
<img width="574" alt="Screen Shot 2023-01-13 at 17 47 00" src="https://user-images.githubusercontent.com/8609783/212289536-0a9fe36d-edaf-479b-bf3c-109aa21bd86e.png">

same issue

**Comment by karlhorky** [comment]
It seems like the new home for the plugin is this, maybe someone wants to report this issue over there as well:

https://github.com/styled-components/typescript-styled-plugin

**Comment by mjbvz** [comment]
Closing as this package has been deprecated in favor of the [official styled-components/typescript-styled-plugin fork]( https://github.com/styled-components/typescript-styled-plugin)

After updating to that fork, follow up in the styled-components repo if this is still an issue
